components/layout/style_ext.rs

```diff
@@ -795,29 +800,40 @@ impl ComputedValuesExt for ComputedValues {
         &self,
         fragment_flags: FragmentFlags,
     ) -> bool {
-        if self.has_transform_or_perspective(fragment_flags) {
-            return true;
-        }
-
-        if !self.get_effects().filter.0.is_empty() {
-            return true;
-        }
-
-        // See <https://drafts.csswg.org/css-transforms-2/#transform-style-property>.
-        if self.is_transformable(fragment_flags) &&
-            self.get_box().transform_style == ComputedTransformStyle::Preserve3d
-        {
-            return true;
-        }
         // From <https://www.w3.org/TR/css-will-change/#valdef-will-change-custom-ident>:
         // > If any non-initial value of a property would cause the element to generate a
         // > containing block for fixed positioned elements, specifying that property in will-change
         // > must cause the element to generate a containing block for fixed positioned elements.
         let will_change_bits = self.clone_will_change().bits;
-        if will_change_bits.intersects(WillChangeBits::FIXPOS_CB_NON_SVG) ||
-            (will_change_bits
-                .intersects(WillChangeBits::TRANSFORM | WillChangeBits::PERSPECTIVE) &&
-                self.is_transformable(fragment_flags))
+
+        // From <https://drafts.csswg.org/css-transforms-1/#transform-rendering>:
+        // > any value other than `none` for the `transform` property also causes the element
+        // > to establish a containing block for all descendants.
+        // From <https://www.w3.org/TR/css-transforms-2/#individual-transforms>
+        // > all other values […] create a stacking context and containing block for all
+        // > descendants, per usual for transforms.
+        // From <https://drafts.csswg.org/css-transforms-2/#perspective-property>:
+        // > The use of this property with any value other than `none` […] establishes a
+        // > containing block for all descendants, just like the `transform` property does.
+        // From <https://drafts.csswg.org/css-transforms-2/#transform-style-property>:
+        // > A computed value of `preserve-3d` for `transform-style` on a transformable element
+        // > establishes both a stacking context and a containing block for all descendants.
```

Comment:
Nit: I would put some newlines here to make this slightly easier to read:

```suggestion
        // > descendants, per usual for transforms.
        //
        // From <https://drafts.csswg.org/css-transforms-2/#perspective-property>:
        // > The use of this property with any value other than `none` […] establishes a
        // > containing block for all descendants, just like the `transform` property does.
        //
        // From <https://drafts.csswg.org/css-transforms-2/#transform-style-property>:
        // > A computed value of `preserve-3d` for `transform-style` on a transformable element
        // > establishes both a stacking context and a containing block for all descendants.
```